installer/scripts/omsadmin.sh

```diff
@@ -591,7 +595,10 @@ onboard()
     if [ $VERBOSE -ne 0 ]; then
         echo
         echo "Generated request:"
-        cat $BODY_ONBOARD
+        # Mask the certificate in the request
+        local body_no_newlines="`cat $BODY_ONBOARD | tr -d '\n' 2> /dev/null`"
+        echo "$body_no_newlines" | sed "s/<AuthenticationCertificate>.*<\/AuthenticationCertificate>/<AuthenticationCertificate>********<\/AuthenticationCertificate>/g"
```

Comment:
You can also use :, or any other character when it is less problematic:
l=AuthenticateCertificate
maskliteral='********'
echo "$body_no_newlines" | sed "s:<$l>[^<]*</$l>:<$l>$maskliteral</$l>:"

And adding the [^<]* wildcard, given sed does NOT have non-greedy *?, prevents you from eating up multiple pairs in case that ever becomes a possibility.

Reply:
I'm using / to remain consistent with the other line in omsadmin.sh that replaces a string with another value: https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/scripts/omsadmin.sh#L1210

As for the local variable and wildcard additions, I will add those.

Reply:
Fixed.